cloud/google/clusteractuator.go

```diff
@@ -60,12 +70,39 @@ func NewClusterActuator(params ClusterActuatorParams) (*GCEClusterClient, error)
 		computeService:         computeService,
 		clusterClient:          params.ClusterClient,
 		gceProviderConfigCodec: codec,
+		eventRecorder:          params.EventRecorder,
 	}, nil
 }
 
 func (gce *GCEClusterClient) Reconcile(cluster *clusterv1.Cluster) error {
 	glog.Infof("Reconciling cluster %v.", cluster.Name)
-	err := gce.createFirewallRuleIfNotExists(cluster, &compute.Firewall{
+	created := gce.createClusterIfNeeded(cluster)
+	if created {
+		gce.eventRecorder.Eventf(cluster, corev1.EventTypeNormal, "Created", "Created Cluster %v", cluster.Name)
+	}
+	return nil
+}
+
+func (gce *GCEClusterClient) Delete(cluster *clusterv1.Cluster) error {
+	err := gce.deleteFirewallRule(cluster, cluster.Name+firewallRuleInternalSuffix)
+	defer gce.fireErrorEvent(cluster, err, deleteClusterEventAction)
```

Comment:
Can you verify that this works as expected? I've definitely written code before that tried to do this and realized that I was miscapturing a variable by not using a pointer. This looks to me like it will always take the value of err from the deleteFirewallRule call and not be updated later in the function. 

Check out https://play.golang.org/p/wwkc2Nr1X35 vs. https://play.golang.org/p/RdshLgxBvsc

Reply:
I didn't know that, thanks! Just fixed here and in createClusterIfNeeded too.